source/slang/parser.cpp

```diff
@@ -2279,7 +2279,7 @@ namespace Slang
         auto nameToken = parser->ReadToken(TokenType::Identifier);
         assocTypeDecl->nameAndLoc = NameLoc(nameToken);
         assocTypeDecl->loc = nameToken.loc;
-        parseOptionalGenericConstraints(parser, assocTypeDecl);
+        parseOptionalInheritanceClause(parser, assocTypeDecl);
```

Comment:
Why this switch? Is there a reason why we prefer to treat the constraints on an associated type as `InheritanceDecl` over `GenericTypeConstraintDecl`? I honestly don't know which is the right choice.